offchain-modules/lib/src/relay/ckb_relay.rs

```diff
@@ -1 +1,70 @@
+use crate::transfer::to_eth::get_add_ckb_headers_abi;
+use crate::util::ckb_util::Generator;
+use crate::util::eth_util::Web3Client;
+use anyhow::Result;
+use ethereum_types::U256;
+use log::*;
+use web3::types::H160;
 
+pub struct CKBRelayer {
+    pub ckb_rpc_url: String,
+    pub indexer_url: String,
+    pub eth_rpc_url: String,
+    pub from: H160,
+    pub contract_addr: H160,
+    pub priv_key_path: String,
+}
+
+impl CKBRelayer {
+    pub fn new(
+        ckb_rpc_url: String,
+        indexer_url: String,
+        eth_rpc_url: String,
+        from: H160,
+        contract_addr: H160,
+        priv_key_path: String,
+    ) -> Self {
+        CKBRelayer {
+            ckb_rpc_url,
+            indexer_url,
+            eth_rpc_url,
+            from,
+            contract_addr,
+            priv_key_path,
+        }
+    }
+    pub async fn start(&mut self) -> Result<()> {
+        let mut ckb_relay = Generator::new(
+            self.ckb_rpc_url.clone(),
+            self.indexer_url.clone(),
+            Default::default(),
+        )
+        .unwrap();
```

Comment:
```suggestion
        .map_err(|e| anyhow::anyhow!("failed to crate generator: {}", e))?;
```

Try not use `unwrap` in production code.

Reply:
> Try not use `unwrap` in production code.

copy that.  I saw that "handler error rather than panic" by mkxbl.